src/lib/io/sol-iio.c

```diff
@@ -366,6 +367,24 @@ set_buffer_enabled(struct sol_iio_device *device, bool enabled)
     return true;
 }
 
+static bool
+set_channel_enabled(struct sol_iio_device *device, const char *channel_name, bool enabled)
+{
+    char path[PATH_MAX];
+    bool r;
+
+    r = craft_filename_path(path, sizeof(path), CHANNEL_ENABLE_PATH, device->device_id, channel_name);
+    if (!r)
+        return false;
+
+    if (check_file_existence(path)) {
```

Comment:
@laykuanloon I would like to suggest you to verify if you're using this call ^ (check_file_existence()) correctly.
It doesn't seem the case to me.
I'm almost sure you just want to call sol_util_write_file() directly after crafting the file name.